### PR TITLE
Fix user search in dev environments

### PR DESCRIPTION
**Pull request by mvandenburgh**
User search doesn't work in local deployments because it only considers `SocialAccounts` (which local environments don't use). This adds an additional query that searches all `User` objects too; the query is conditionalized so it only runs when the `SocialAccount` search returns no results (otherwise there would be duplicates between the `User` and `SocialAccount` objects).

This also allows the E2E tests to test adding an owner to a dandiset, which will hopefully prevent that feature from breaking again.

### CODE REVIEW

File: dandiapi/api/views/users.py [hunk 1]
```diff
@@ -78,5 +78,15 @@ def users_search_view(request: Request) -> HttpResponseBase:
     social_accounts = SocialAccount.objects.filter(extra_data__icontains=username)[:10]
     users = [social_account_to_dict(social_account) for social_account in social_accounts]
 
+    # Try searching Django's regular `User`s if there aren't any results.
+    # This allows this feature to work in development without having to conditionalize
+    # code based on the type of deployment.
+    if len(users) == 0:
```
Comment (dchiquito) [inline dandiapi/api/views/users.py:84]:
```suggestion
    if not users:
```
Apparently this is more pythonic https://stackoverflow.com/a/53522

File: dandiapi/api/views/users.py [hunk 2]
```diff
@@ -78,5 +78,15 @@ def users_search_view(request: Request) -> HttpResponseBase:
...
+        users = [
+            user_to_dict(user)
+            for user in User.objects.filter(username__icontains=username)[:10]
+            if user.username != 'AnonymousUser'
```
Comment (dchiquito) [inline dandiapi/api/views/users.py:88]:
```suggestion
            for user in User.objects.filter(username__icontains=username).filter(~Q(username='AnonymousUser'))[:10]
```
You will also need `from django.db.models import Q`

File: dandiapi/api/views/users.py (hunk 1 above)
Comment (mvandenburgh) [reply dandiapi/api/views/users.py]: Changed it

File: dandiapi/api/views/users.py (hunk 2 above)
Comment (mvandenburgh) [reply dandiapi/api/views/users.py]: Changed it

File: dandiapi/api/views/users.py [hunk 3]
```diff
@@ -78,5 +79,16 @@ def users_search_view(request: Request) -> HttpResponseBase:
     social_accounts = SocialAccount.objects.filter(extra_data__icontains=username)[:10]
     users = [social_account_to_dict(social_account) for social_account in social_accounts]
 
+    # Try searching Django's regular `User`s if there aren't any results.
+    # This allows this feature to work in development without having to conditionalize
+    # code based on the type of deployment.
+    if not users == 0:
```
Comment (dchiquito) [inline dandiapi/api/views/users.py:85]:
```suggestion
    if not users:
```

**Review by dchiquito — CHANGES_REQUESTED** [review]

File: dandiapi/api/views/users.py (hunk 3 above)
Comment (mvandenburgh) [reply dandiapi/api/views/users.py]: :facepalm:

**Review by dchiquito — APPROVED** [review]

**Review by waxlamp — COMMENTED** [review]
Nice bugfix. I have a question: why not always search for `User` models? What's the benefit in looking at `SocialAccount` first, if the same results would show up in `User` as well?

**Comment by mvandenburgh** [comment]
> Nice bugfix. I have a question: why not always search for `User` models? What's the benefit in looking at `SocialAccount` first, if the same results would show up in `User` as well?

Using `SocialAccount` allows searching by a user's github username.

**Pull request merged**